app/controllers/api/canvas_users_controller.rb

```diff
@@ -22,4 +17,83 @@ def index
       status: :ok,
     )
   end
+
+  def update
+    edit_user_response = edit_user_on_canvas
+
+    # This ID is the ID of the login record; it's different from the user's login ID.
+    numeric_login_id = find_canvas_user_login["id"]
+
+    edit_user_login_response = edit_user_login_on_canvas(numeric_login_id)
+
+    render(
+      json: {
+        id: params[:id],
+        name: edit_user_response["name"],
+        login_id: edit_user_login_response["unique_id"],
+        sis_user_id: edit_user_login_response["sis_user_id"],
+        email: edit_user_response["email"],
+      },
+      status: :ok,
+    )
+  end
+
+  private
+
+  def validate_current_user_lti_admin
+    unless current_user.lti_admin?(jwt_context_id)
+      user_not_authorized "Only account admins are authorized to use this application."
+    end
+  end
+
+  def validate_user_is_in_account
+    user_is_in_account = search_for_users_on_canvas(params[:id]).
+      parsed_response.
+      present?
+
+    unless user_is_in_account
+      user_not_authorized(
+        "You are only authorized to modify users from the account or sub-accounts you administer.",
+      )
+    end
+  end
+
+  def search_for_users_on_canvas(search_term, page = nil)
+    # We're manually constructing the URL here and using `api_get_request` instead of
+    # `canvas_api.proxy("LIST_USERS_IN_ACCOUNT", params)` because `proxy("LIST_USERS_IN_ACCOUNT")`
+    # doesn't support the `include` param since it's undocumented.
+    canvas_url = "accounts/#{params[:canvas_account_id]}/users" \
+      "?search_term=#{search_term}&include[]=email"
+    canvas_url += "&page=#{page}" if page # You get a 404 if you pass `&page=`
```

Comment:
Consider using `to_query`?
```suggestion
    query_params = {
      search_term: search_term,
      include: [:email],
    }
    query_params[:page] = page if page.present? # You get a 404 if you pass `&page=`
    account_id = params[:canvas_account_id]
    canvas_url = "accounts/#{account_id}/users?#{query_params.to_query}"
```

Reply:
Yeah, that's a nice improvement. Thanks. I'll add it.

Reply:
Added